src/components/username-input.js

```diff
@@ -0,0 +1,58 @@
+import React from 'react';
+import autoBind from 'auto-bind';
+import PropTypes from 'prop-types';
+import { Form, Input } from 'antd';
+import { withUser } from './user-context';
+import { withTranslation } from 'react-i18next';
+import { userProps, translationProps, formItemLayoutShape } from '../ui/default-prop-types';
+
+const FormItem = Form.Item;
+
+class UsernameInput extends React.Component {
+  constructor(props) {
+    super(props);
+    autoBind(this);
+  }
+
+  render() {
+    const { forbiddenUsernames, formItemLayout, user, t } = this.props;
+
+    const validationRules = [
+      {
+        required: true,
+        message: t('enterUsername'),
+        whitespace: true
+      },
+      {
+        validator: (rule, value) => {
+          const minLength = 6;
+          return value && value.trim().length < minLength
```

Comment:
Would value?.trim().length < minLength work?

Reply:
Actually this doesn't work because we need (A and B) and not just (B) in this condition.
So we should only check the length when a value actually exists.